programs/pdg/write.ts

```diff
@@ -1,42 +1,162 @@
+  .option('-wc, --withdrawalCredentials [wc]', 'withdrawal credentials')
+  .argument('<deposits>', 'deposits', parseDepositArray)
+  .action(
+    async (
+      deposits: Deposit[],
+      options: { vault: Address; withdrawalCredentials: Hex },
+    ) => {
+      const vault = options.vault;
+      let withdrawalCredentials = options.withdrawalCredentials;
+
+      if (!vault && !withdrawalCredentials) {
+        throw new Error(
+          'You must provide either vault or withdrawal credentials',
+        );
+      } else if (vault && withdrawalCredentials) {
+        throw new Error(
+          'You can only provide one of vault or withdrawal credentials',
+        );
+      }
+      const bls = getBLSHarnessContract();
+      let hideSpinner = showSpinner({
+        type: 'bouncingBar',
+        message: 'Loading metadata...',
+      });
+      const pdg = await getPredepositGuaranteeContract();
+      const PREDEPOSIT_AMOUNT = await callReadMethod(pdg, 'PREDEPOSIT_AMOUNT');
+      if (!PREDEPOSIT_AMOUNT) return;
+
+      if (vault) {
+        const vaultContract = getStakingVaultContract(vault);
+        const wc = await callReadMethod(vaultContract, 'withdrawalCredentials');
+        if (!wc) return;
+        withdrawalCredentials = wc;
+      }
+      hideSpinner();
+
+      for (const deposit of deposits) {
+        // amount check
+        if (deposit.amount !== PREDEPOSIT_AMOUNT) {
+          console.info(
+            `❌ Deposit amount is not equal to PREDEPOSIT_AMOUNT for pubkey ${deposit.pubkey}`,
+          );
+        } else {
+          console.info(`✅ AMOUNT VALID for Pubkey ${deposit.pubkey}`);
+        }
+
+        // depositDataRoot check
+        const depositDataRoot = computeDepositDataRoot(
+          deposit.pubkey,
+          withdrawalCredentials,
+          deposit.signature,
+          deposit.amount,
+        );
+        if (depositDataRoot != deposit.depositDataRoot) {
+          console.info(
+            `❌ depositDataRoot does not match ${deposit.pubkey}, actual root: ${depositDataRoot}`,
+          );
+        } else {
+          console.info(`✅ depositDataRoot VALID for Pubkey ${deposit.pubkey}`);
+        }
+
+        // local BLS check
+        const isBLSValid = isValidBLSDeposit(deposit, withdrawalCredentials);
+        if (!isBLSValid) {
+          console.info(
+            `❌ Offchain - BLS signature is not valid for Pubkey ${deposit.pubkey}`,
+          );
+        } else {
+          console.info(`✅ SIGNATURE VALID for Pubkey ${deposit.pubkey}`);
+        }
+
+        // onchain BLS check
+        const {
+          pubkeyY_a,
+          pubkeyY_b,
+          sigY_c0_a,
+          sigY_c0_b,
+          sigY_c1_a,
+          sigY_c1_b,
+        } = expandBLSSignature(deposit.signature, deposit.pubkey);
+        hideSpinner = showSpinner({
+          type: 'bouncingBar',
+          message: 'Checking onchain againts BLSHarness contract',
+        });
+        const isValid = await bls.read
```

Comment:
callReadMethod

Reply:
However, if the function's result is inappropriate here, you can leave it as is for future improvements.